apps/gateway-ui/src/App.tsx

```diff
@@ -50,18 +50,24 @@ export const App = React.memo(function Admin(): JSX.Element {
           setLoading(false);
         });
     } else {
-      gateway
-        .fetchInfo()
-        .then((gatewayInfo) => {
-          setGatewayInfo(gatewayInfo);
-        })
-        .catch(({ message, error }) => {
-          console.error(error);
-          setError(message);
-        })
-        .finally(() => {
-          setLoading(false);
-        });
+      const fetchInfo = () => {
+        gateway
+          .fetchInfo()
+          .then((gatewayInfo) => {
+            setGatewayInfo(gatewayInfo);
+          })
+          .catch(({ message, error }) => {
+            console.error(error);
+            setError(message);
+          })
+          .finally(() => {
```

Comment:
nit: We probably should remove `setLoading` state call from the polling fn

Reply:
fixed.